consensus/spos/bls/v2/subroundSignature.go

```diff
@@ -260,44 +176,20 @@ func (sr *subroundSignature) doSignatureConsensusCheck() bool {
 		return true
 	}
 
-	// TODO[cleanup cns finality]: simply return false and remove the rest of the method. This will be handled by subroundEndRound
-	if sr.EnableEpochsHandler().IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, sr.GetHeader().GetEpoch()) {
-		return false
+	selfJobDone := true
+	if sr.IsNodeInConsensusGroup(sr.SelfPubKey()) {
+		selfJobDone = sr.IsSelfJobDone(sr.Current())
 	}
-
-	isSelfLeader := sr.IsSelfLeader()
-
-	threshold := sr.Threshold(sr.Current())
-	if sr.FallbackHeaderValidator().ShouldApplyFallbackValidation(sr.GetHeader()) {
-		threshold = sr.FallbackThreshold(sr.Current())
-		log.Warn("subroundSignature.doSignatureConsensusCheck: fallback validation has been applied",
-			"minimum number of signatures required", threshold,
-			"actual number of signatures received", sr.getNumOfSignaturesCollected(),
-		)
+	multiKeyJobDone := true
+	if sr.IsMultiKeyInConsensusGroup() {
+		multiKeyJobDone = sr.IsMultiKeyJobDone(sr.Current())
 	}
 
-	areSignaturesCollected, numSigs := sr.areSignaturesCollected(threshold)
-	areAllSignaturesCollected := numSigs == sr.ConsensusGroupSize()
-
-	isSignatureCollectionDone := areAllSignaturesCollected || (areSignaturesCollected && sr.GetWaitingAllSignaturesTimeOut())
-	isJobDoneByLeader := isSelfLeader && isSignatureCollectionDone
-
-	isSelfJobDone := sr.IsSelfJobDone(sr.Current())
-	isJobDoneByConsensusNode := !isSelfLeader && isSelfInConsensusGroup && isSelfJobDone
-
-	isSubroundFinished := isJobDoneByConsensusNode || isJobDoneByLeader
-
-	if isSubroundFinished {
-		if isSelfLeader {
-			log.Debug("step 2: signatures",
-				"received", numSigs,
-				"total", len(sr.ConsensusGroup()))
-		}
-
+	isJobDoneByConsensusNode := isSelfInConsensusGroup && selfJobDone && multiKeyJobDone
```

Comment:
isSelfInConsensusGroup can be removed, it will already return on L176 if the current node is not in consensus

Reply:
done